app/views/agents/show.html.erb

```diff
@@ -161,17 +161,19 @@
               <b>Options:</b>
               <pre><%= Utils.pretty_jsonify @agent.options || {} %></pre>
             </p>
-
-            <p id="memory" data-agent-id="<%= @agent.id %>">
-              <b>Memory:</b>
-              <% if @agent.memory.present? %>
-                <i class="fa fa-spinner fa-pulse spinner"></i>
-                <i class="fa fa-trash action-icon clear"></i>
-              <% end %>
-              <pre class="memory"><%= Utils.pretty_jsonify @agent.memory || {} %></pre>
-            </p>
+            <button type="submit" class="btn btn-default" id="target">Hide or Reveal Memory</button>
```

Comment:
Can you rename `target` to something more descriptive, like `toggle-memory`?  The text could be "Show Memory" and the CoffeeScript above could be

```coffee
$('#toggle-memory').click ->
  if $('#toggle-memory').text() == 'Show Memory"
    $('#toggle-memory').text("Hide Memory")
    $('#memory').removeClass 'hidden'
  else
    $('#toggle-memory').text("Show Memory")
    $('#memory').addClass 'hidden'
```

Finally, you don't need the `memorytoggle` div, since you can target `#memory` directly, just add the `hidden` class to it to start.